q2_gunc/assets/js/scripts.js

```diff
@@ -245,18 +253,53 @@ document.addEventListener('DOMContentLoaded', function() {
     option.textContent = sampleId === "" ? "(Default)" : sampleId;
     sampleSelector.appendChild(option);
   });
-  // Default to showing all samples (empty value = all samples)
-  sampleSelector.value = "";
-  magSelector.disabled = true;
+  
+  // For FeatureData[MAG], disable sample selector and enable MAG selector
+  // For SampleData[MAGs], default to showing all samples
+  if (isFeatureDataMAG) {
+    sampleSelector.value = "";
+    sampleSelector.disabled = true;
+    magSelector.disabled = false;
+    // Populate MAGs immediately for FeatureData[MAG]
+    populateMAGs("");
+  } else {
+    sampleSelector.value = "";
+    sampleSelector.disabled = false;
+    magSelector.disabled = true;
+  }
   // Function to populate MAG dropdown
   function populateMAGs(selectedSample) {
     magSelector.innerHTML = '<option value="">-- Select a MAG --</option>';
     plotCard.style.display = 'none';
-    // Handle "All Samples" selection (empty value)
+    
+    // Determine if this is FeatureData[MAG] (dereplicated MAGs)
+    const isFeatureDataMAG = Object.keys(samples).includes("") && Object.keys(samples).length === 1;
```

Comment:
Again, why do we need to calculate this here one more time? Can't we just assume that if "" was passed then it's feature data and otherwise sample data? If not, then perhaps at least we can pass this variable to the function rather then calculate again.